### PR TITLE
fix: Fix case where df.peek would fail to execute even with force=True

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/core/tree_properties.py [hunk 1]
```diff
@@ -25,3 +29,12 @@ def is_trivially_executable(node: nodes.BigFrameNode) -> bool:
 
 def local_only(node: nodes.BigFrameNode) -> bool:
     return all(isinstance(node, nodes.ReadLocalNode) for node in node.roots)
+
+
+@functools.cache
```
Comment (tswast) [inline bigframes/core/tree_properties.py:34]: Would this get unwieldy with local data? I worry about that Feather data getting stored forever.

Comment (TrevorBergeron) [reply bigframes/core/tree_properties.py]: Yeah, not sure why I put that cache decorator. Would take a really long time for the cache to get too big though. Removed and will figure out the whole property caching thing later.

**Review by tswast — COMMENTED** [review]
I'm worried about cached behavior and peelable getting out of sync again. Any chance we could refactor to avoid the possibility of the null value after we call cached?

**Comment by TrevorBergeron** [comment]
> I'm worried about cached behavior and peelable getting out of sync again. Any chance we could refactor to avoid the possibility of the null value after we call cached?

Added 

> I'm worried about cached behavior and peelable getting out of sync again. Any chance we could refactor to avoid the possibility of the null value after we call cached?

Now forcing the materialization after caching.

**Review by tswast — APPROVED** [review]

**Pull request merged**